#!/usr/bin/env python
import math
import os.path
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("position_x", double_t, 0, "x coordinate (in m) of position of axle in base frame", default=0.0)
gen.add("position_y", double_t, 0, "y coordinate (in m) of position of axle in base frame", default=0.0)
gen.add("wheel_diameter", double_t, 0, "in m", default=0.0, min=0.0)
gen.add("is_steered", bool_t, 0, "", default=False)
gen.add("is_driven", bool_t, 0, "", default=False)
gen.add("track", double_t, 0, "distance (in m) between left and right wheel", default=0.0, min=0.0)
gen.add("steering_hinge_offset", double_t, 0,
        "offset (in m) between steering hinge of each wheel and the wheel's center (only used if steered and driven)",
        default=0.0, min=0.0)
gen.add("steering_velocity", double_t, 0, "expected change of steering position per second", default=0.0, min=0.0)
gen.add("steering_position_tolerance", double_t, 0, default=0.001, min=0.0, max=round(math.pi / 2.0, 4),
        description="change in steering position below which steering motors are not expected to move")
gen.add("left_hinge_joint", str_t, 0, "", default="")
gen.add("left_wheel_joint", str_t, 0, "", default="")
gen.add("right_hinge_joint", str_t, 0, "", default="")
gen.add("right_wheel_joint", str_t, 0, "", default="")

package_name = os.path.basename(os.path.dirname(os.path.dirname(__file__)))
node_name = package_name + "_node"
cfg_name = os.path.splitext(os.path.basename(__file__))[0]
exit(gen.generate(package_name, node_name, cfg_name))
